File: native-android/app/src/main/java/site/smap/gugudan/core/adventure/Shop.kt
package site.smap.gugudan.core.adventure

// 꾸미기 상점 카탈로그 (Shop.swift 이식 — 가격/구성 동일)

data class ShopColor(val id: String, val name: String, val hex: String, val price: Int)
data class ShopHat(val id: String, val name: String, val emoji: String, val price: Int)

object Shop {
    const val DEFAULT_COLOR_ID = "indigo"

    val colors: List<ShopColor> = listOf(
        ShopColor("indigo", "인디고", "#6366f1", 0),
        ShopColor("rose", "로즈", "#f43f5e", 8),
        ShopColor("amber", "앰버", "#f59e0b", 8),
        ShopColor("emerald", "에메랄드", "#10b981", 8),
        ShopColor("sky", "하늘", "#0ea5e9", 8),
        ShopColor("violet", "바이올렛", "#8b5cf6", 8),
    )

    val hats: List<ShopHat> = listOf(
        ShopHat("sprout", "새싹", "🌱", 10),
        ShopHat("straw", "밀짚모자", "👒", 15),
        ShopHat("tophat", "신사 모자", "🎩", 25),
        ShopHat("crown", "황금 왕관", "👑", 40),
    )

    fun color(id: String): ShopColor? = colors.firstOrNull { it.id == id }
    fun hat(id: String): ShopHat? = hats.firstOrNull { it.id == id }
}

File: native-ios/Gugu/Core/Adventure/Shop.swift
import Foundation

// 꾸미기 상점 — 별 조각으로 구매하는 플레이어 커스터마이징 카탈로그 (순수 데이터, UI 무의존)

struct ShopColor: Identifiable, Equatable {
    let id: String
    let name: String
    let hex: String
    let price: Int   // 0 = 기본 제공
}

struct ShopHat: Identifiable, Equatable {
    let id: String
    let name: String
    let emoji: String  // 2D 프리뷰/목록용
    let price: Int
}

enum Shop {
    static let defaultColorID = "indigo"

    static let colors: [ShopColor] = [
        ShopColor(id: "indigo", name: "인디고", hex: "#6366f1", price: 0),
        ShopColor(id: "rose", name: "로즈", hex: "#f43f5e", price: 8),
        ShopColor(id: "amber", name: "앰버", hex: "#f59e0b", price: 8),
        ShopColor(id: "emerald", name: "에메랄드", hex: "#10b981", price: 8),
        ShopColor(id: "sky", name: "하늘", hex: "#0ea5e9", price: 8),
        ShopColor(id: "violet", name: "바이올렛", hex: "#8b5cf6", price: 8),
    ]

    static let hats: [ShopHat] = [
        ShopHat(id: "sprout", name: "새싹", emoji: "🌱", price: 10),
        ShopHat(id: "straw", name: "밀짚모자", emoji: "👒", price: 15),
        ShopHat(id: "tophat", name: "신사 모자", emoji: "🎩", price: 25),
        ShopHat(id: "crown", name: "황금 왕관", emoji: "👑", price: 40),
    ]

    static func color(_ id: String) -> ShopColor? { colors.first { $0.id == id } }
    static func hat(_ id: String) -> ShopHat? { hats.first { $0.id == id } }
}
